Configuration/EventContent/python/EventContent_cff.py

```diff
@@ -620,6 +620,10 @@ def SwapKeepAndDrop(l):
 FEVTDEBUGEventContent.outputCommands.extend(SimMuonFEVTDEBUG.outputCommands)
 FEVTDEBUGEventContent.outputCommands.extend(SimCalorimetryFEVTDEBUG.outputCommands)
 FEVTDEBUGEventContent.outputCommands.extend(SimFastTimingFEVTDEBUG.outputCommands)
+FEVTDEBUGEventContent.outputCommands.append('keep *_muons1stStep_*_*')
```

Comment:
In the PR description, it appears that all added collections correspond to `*_muons1stStep_*`
What about the following three append statements? 
Were they supposed to add collections to the output?

Reply:
Hi @mandrenguyen I tested everything again and indeed the other collections are not needed.  `*_muon1stStep_*` is also not fully needed, only `recoMuons_muon1stStep_*` is necessary for using `customiseTICLFromReco` .

Reply:
Thanks.  But why is `keep recoMuons_muon1stStep_*_*` necessary
One would have expected that this is already included in any `FEVT` content. 
I think I'm missing something here.

Reply:
`customiseTICLFromReco` is used to be able to run the TICL reconstruction starting from a step3 output file. 
Without the `keep recoMuons_muon1stStep_*_*` this is the event content in a step3.root file: [eventContentStep3](https://wredjeb.web.cern.ch/FixTICLFromReco/eventContentStep3.log) and as you can see the `recoMuons_muon1stStep_*_*` collection is not there. When you try to run a step3 with `customiseTICLFromReco` you get the following error:
```
----- Begin Fatal Exception 21-Jun-2023 14:06:18 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Prefetching for module HGCalValidator/'hgcalValidator'
   [2] Calling method for module TrackstersMergeProducer/'ticlTrackstersMerge'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Muon>
Looking for module label: muons1stStep
Looking for productInstanceName: 

----- End Fatal Exception -------------------------------------------------
```

Reply:
Ok, I think muons1stStep is normally not stored here:
https://github.com/cms-sw/cmssw/blob/master/RecoMuon/Configuration/python/RecoMuon_EventContent_cff.py

For my understanding, why does tickle consume `muons1stStep` instead of`muons`, which is already stored?
Do we really want to change FEVT for every workflow, including Phase 1?

Reply:
Just in case we will change FEVT, the line should not be here. I think it fits better in 
https://github.com/cms-sw/cmssw/blob/master/RecoMuon/Configuration/python/RecoMuon_EventContent_cff.py#L98-L103

Reply:
@mandrenguyen If I understood correctly, TICL consumes `muons1stStep` because the `muons` collection is not yet there, since it needs the Particle Flow reconstruction that comes after TICL. As a quick test: trying to make TICL consume `muons` instead of `muons1stStep` you get the following error:
```
----- Begin Fatal Exception 21-Jun-2023 16:58:36 CEST-----------------------
An exception of category 'ScheduleExecutionFailure' occurred while
   [0] Calling beginJob
Exception Message:
Unrunnable schedule
Circular module dependency found in configuration
  module 'CSCTightHaloFilter' depends on BeamHaloSummary
  module 'BeamHaloSummary' depends on CSCHaloData
  module 'CSCHaloData' depends on muons
  module 'muons' depends on particleFlowTmp
  module 'particleFlowTmp' depends on particleFlowTmpBarrel
  module 'particleFlowTmpBarrel' depends on particleFlowBlock
  module 'particleFlowBlock' depends on pfTrackElec
  module 'pfTrackElec' depends on electronGsfTracks
  module 'electronGsfTracks' depends on electronCkfTrackCandidates
  module 'electronCkfTrackCandidates' depends on electronMergedSeeds
  module 'electronMergedSeeds' depends on ecalDrivenElectronSeeds
  module 'ecalDrivenElectronSeeds' depends on particleFlowSuperClusterHGCal
  module 'particleFlowSuperClusterHGCal' depends on particleFlowClusterHGCal
  module 'particleFlowClusterHGCal' depends on ticlTrackstersMerge
  module 'ticlTrackstersMerge' depends on muons
----- End Fatal Exception -------------------------------------------------
```

It is true that the `muons` collection is saved in step3.root, but we don't want to modify the TICL RE-RECO (using `customiseTICLFromReco`) consuming different objects.

Reply:
I see the point.  Is the re-TICL from RECO only ever intended to run on the heavy FEVT content?
Because if you even intend to run it on a lighter event content such as AOD, the overhead of storing an additional muon collection is going to be large.

Reply:
Hi @mandrenguyen no we won't run it on AOD but only on the FEVT content. 

I proposed a change [below](https://github.com/cms-sw/cmssw/pull/42016#discussion_r1237270029) to avoid affectting Phase-1 workflows, please let me know if it works.

Reply:
Let's please address this comment about moving the append line:
https://github.com/cms-sw/cmssw/pull/42016#discussion_r1237019602
and please post the overall change to the compressed event size due to this append statement.